portal/src/main/webapp/WEB-INF/jsp/patient_view/mdacc_patient_heatmap_viewer.jsp

```diff
@@ -0,0 +1,104 @@
+<%--
+ - Copyright (c) 2016 Memorial Sloan-Kettering Cancer Center.
+ -
+ - This library is distributed in the hope that it will be useful, but WITHOUT
+ - ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS
+ - FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+ - is on an "as is" basis, and Memorial Sloan-Kettering Cancer Center has no
+ - obligations to provide maintenance, support, updates, enhancements or
+ - modifications. In no event shall Memorial Sloan-Kettering Cancer Center be
+ - liable to any party for direct, indirect, special, incidental or
+ - consequential damages, including lost profits, arising out of the use of this
+ - software and its documentation, even if Memorial Sloan-Kettering Cancer
+ - Center has been advised of the possibility of such damage.
+ --%>
+
+<%--
+ - This file is part of cBioPortal.
+ -
+ - cBioPortal is free software: you can redistribute it and/or modify
+ - it under the terms of the GNU Affero General Public License as
+ - published by the Free Software Foundation, either version 3 of the
+ - License.
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program.  If not, see <http://www.gnu.org/licenses/>.
+--%>
+
+
+<%--  For the PATIENT view  --%>
+
+<%@ page import="org.mskcc.cbio.portal.util.GlobalProperties" %>
+
+
+<script type="text/javascript">
+
+    $(function () {
+
+        /*
+         * <li id="tab-heatmap-viewer"><a id="link-heatmap-viewer" href="#tab_chm" class="patient-tab">Heatmap</a></li>
+         * _and_
+         * <div class="patient-section" id="tab_chm">
+         *     etc etc
+         * </div>
+         */
+
+        var tabLI = $('#tab-heatmap-viewer'),             // the <li> for the tab
```

Comment:
I made a comparison. I think only this line of code ```if (!$(this).parent().hasClass('ui-state-disabled') && !$(this).hasClass("tab-clicked"))``` cannot be merged 3(may be include the whole click function as a parameter). Other differences can all be passed as parameters if we make code as a stand alone module, it would be easier to modify in the future.